src/docfx/lib/HtmlUtility.cs

```diff
@@ -325,6 +325,44 @@ public static string CreateHtmlMetaTags(JObject metadata)
         return result.ToString();
     }
 
+    public static void CollectHtmlUsage(string html, Dictionary<string, Dictionary<string, int>> elementCount)
+    {
+        var reader = new HtmlReader(html);
+
+        while (reader.Read(out var token))
+        {
+            CollectHtmlUsageCore(ref token, elementCount);
+        }
+
+        static void CollectHtmlUsageCore(ref HtmlToken token, Dictionary<string, Dictionary<string, int>> elementCount)
+        {
+            if (token.Type != HtmlTokenType.StartTag)
+            {
+                return;
+            }
+
+            var tokenName = token.Name.ToString();
+            if (!elementCount.ContainsKey(tokenName))
+            {
+                elementCount.Add(tokenName, new Dictionary<string, int>());
+            }
+
+            var attributeCount = elementCount[tokenName];
+            if (token.Attributes.Span.IsEmpty)
+            {
+                attributeCount[string.Empty] = attributeCount.GetValueOrDefault(string.Empty, 0) + 1;
```

Comment:
Use `CollectionsMarshal` to same dictionary lookups. Same in line 360.
```suggestion
                CollectionsMarshal.GetValueRefOrAddDefault(attributeCount, "", out _)++;
```